http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -849,6 +851,34 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
         )
     } yield ()
 
+  private[monitor] def handleDeleteKubernetesClusterMessage(msg: DeleteKubernetesClusterMessage)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      // TODO: ${ctx.traceId} ends up being None with manually published messages. Make sure that's not the case with cron-job-created messages
+      ctx <- ev.ask
+      clusterId = msg.clusterId
+      _ <- logger.info(
+        s"Beginning clean-up of cluster $clusterId in project ${msg.project} because it has had no apps running for a period of time. | trace id: ${ctx.traceId}"
+      )
+      // TODO: Should we check again that the cluster is okay to delete in case a user requested app creation since the cron job published the message?
+      _ <- kubernetesClusterQuery.markPendingDeletion(clusterId).transaction
+      _ <- gkeInterp
+      // TODO: Should we retry failures and with what RetryConfig? If all retries fail, send an alert?
```

Comment:
I think wb-libs retries internally? Don't think we need an extra layer of retries in Leo.